tests/testthat/test_reset.R

```diff
@@ -2,12 +2,281 @@ context("reset.R functions")
+        test_t_reset <- resetDGEobj(test_t_obj)
+        expect_true("design" %in% names(test_t_reset)) # check if persists
+
+        # testing t_obj with add item
+        test_t_obj <- addItem(test_t_obj,
+                              item = 'Fred Flintstone',
+                              itemName = 'Cartoon',
+                              itemType = 'meta',
+                              itemAttr = list('MyAttribute' = 'testObject'))
+        expect_true("Cartoon" %in% names(test_t_obj))
+        test_t_reset <- resetDGEobj(test_t_obj)
+        expect_false("Cartoon" %in% names(test_t_reset)) # check if removed
+
+        # testing t_obj after class change
+        test_t_obj <- as.list.DGEobj(test_t_obj)
+        expect_equal(class(test_t_obj), "list")
+        # coerce back into DGEobj, expect reset to work
+        class(test_t_obj) <- "DGEobj"
+        expect_silent(resetDGEobj(test_t_obj))
+    }
+
+    expect_success(test_gene_obj(t_obj))
+
+    # test isoform level data
+    test_isoform_obj <- function(isoform_data) {
+
+        test_t_obj <- isoform_data
+        test_t_reset <- resetDGEobj(test_t_obj)
+
+        # object validation
+        expect_s3_class(test_t_reset, "DGEobj")
+        expect_equivalent(showMeta(test_t_reset), showMeta(test_t_obj))
+
+        # validate level is isoform before and after reset
+        test_t_meta <- showMeta(test_t_obj)
+        expect_equal("isoform", test_t_meta$Value[3])
+        expect_equal(attr(test_t_obj, "level"), "isoform")
+
+        test_t_meta_reset <- showMeta(test_t_reset)
+        expect_equal("isoform", test_t_meta_reset$Value[3])
+        expect_equal(attr(test_t_reset, "level"), "isoform")
+
+        # reset after subsetting
+        test_t_dim <- dim(test_t_obj)
+        expect_equal(dim(test_t_obj),c(951, 48))
+        test_t_obj <- test_t_obj[c(1:10), ]
+        test_t_reset <- resetDGEobj(test_t_obj)
+        expect_equal(dim(test_t_reset), c(1000, 48))
+
+        # check names after reset
+        expect_named(test_t_obj, c('counts_orig', 'counts', 'design_orig', 'design', 'isoformData_orig',
+                                   'isoformData', 'granges_orig', 'granges', 'DGEList', 'ReplicateGroupDesign',
+                                   'ReplicateGroupDesign_Elist', 'ReplicateGroupDesign_fit',
+                                   'ReplicateGroupDesign_fit_cm', 'ReplicateGroupDesign_fit_cf',
+                                   'BDL_vs_Sham', 'EXT1024_vs_BDL', 'Nint_vs_BDL', 'Sora_vs_BDL'))
+        expect_named(test_t_reset, c("counts_orig", "counts", "design_orig", "design",
+                                     "isoformData_orig", "isoformData", "granges_orig", "granges"))
+
+        # testing t_obj with rm item
+        test_t_obj <- rmItem(test_t_obj, "design")
+        expect_false("design" %in% names(test_t_obj))
+        test_t_reset <- resetDGEobj(test_t_obj)
+        expect_true("design" %in% names(test_t_reset)) # check if persists
+
```

Comment:
Remove something other than design - each test can remove something different and then your test suite is that much richer.  For instance, remove a contrast from the object (ie BDL_vs_Sham) -- it won't be restored.  Or remove the "intensities" which should be restored, etc.  Diversifing so that you aren't just running the same test over and over will help your suite and the team.

Reply:
Even moving this to isoformData would be better - "design" being removed on all the objects is pretty boring, and that's fine if you want to test it but there is no difference in functinality between these objects and the design field - the difference comes in the "levelData" fields and the primaryAssay fields.